File: apply-migration-direct.js
#!/usr/bin/env node

/**
 * 🔧 APLICAR MIGRATION: Corrigir RLS para Basic Supabase Authentication
 */

import { createClient } from '@supabase/supabase-js';

const SUPABASE_URL = 'https://marajvabdwkpgopytvhh.supabase.co';
const SUPABASE_SERVICE_ROLE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im1hcmFqdmFiZHdrcGdvcHl0dmhoIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc0OTc2NDAwOSwiZXhwIjoyMDY1MzQwMDA5fQ.mkQBVPvhM3OJndsyinoONRUHSDJMh1nFBbPPNH_6cYY';

const supabase = createClient(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY);

async function aplicarMigration() {
  console.log('🔧 APLICANDO MIGRATION: Basic Supabase Authentication');
  console.log('===================================================');

  try {
    // Verificar tabela existe
    console.log('\n📋 Verificando tabela pipeline_outcome_reasons...');
    
    const { data: tableTest, error: tableError } = await supabase
      .from('pipeline_outcome_reasons')
      .select('id')
      .limit(1);
      
    if (tableError) {
      console.error('❌ Erro acessando tabela:', tableError.message);
      return;
    }
    
    console.log('✅ Tabela pipeline_outcome_reasons acessível');
    console.log(`📊 Teste de acesso: ${tableTest?.length || 0} registros visíveis`);

    console.log('\n💡 MIGRATION DEVE SER APLICADA VIA SUPABASE DASHBOARD:');
    console.log('=====================================================');
    console.log('');
    console.log('1. 🌐 Acesse: https://supabase.com/dashboard/project/marajvabdwkpgopytvhh');
    console.log('2. 📊 Vá em: Database > SQL Editor');
    console.log('3. 📝 Copie e execute o SQL abaixo:');
    console.log('');
    console.log('-- ============================================');
    console.log('-- 🔧 MIGRATION: Basic Supabase Authentication');
    console.log('-- ============================================');
    console.log('');
    console.log('-- ETAPA 1: Remover políticas antigas');
    console.log('DROP POLICY IF EXISTS "Users can view outcome reasons for their tenant" ON pipeline_outcome_reasons;');
    console.log('DROP POLICY IF EXISTS "Admins can manage outcome reasons for their tenant" ON pipeline_outcome_reasons;');
    console.log('DROP POLICY IF EXISTS "Users can view outcome history for their tenant" ON lead_outcome_history;');
    console.log('DROP POLICY IF EXISTS "Users can create outcome history for their tenant" ON lead_outcome_history;');
    console.log('');
    console.log('-- ETAPA 2: Criar políticas Basic Supabase Auth');
    console.log('CREATE POLICY "basic_auth_view_outcome_reasons"');
    console.log('  ON pipeline_outcome_reasons');
    console.log('  FOR SELECT');
    console.log('  USING (');
    console.log('    auth.uid() IS NOT NULL');
    console.log('    AND tenant_id = (');
    console.log('      SELECT user_metadata->>\'tenant_id\'');
    console.log('      FROM auth.users');
    console.log('      WHERE id = auth.uid()');
    console.log('    )');
    console.log('  );');
    console.log('');
    console.log('CREATE POLICY "basic_auth_manage_outcome_reasons"');
    console.log('  ON pipeline_outcome_reasons');
    console.log('  FOR ALL');
    console.log('  USING (');
    console.log('    auth.uid() IS NOT NULL');
    console.log('    AND tenant_id = (');
    console.log('      SELECT user_metadata->>\'tenant_id\'');
    console.log('      FROM auth.users');
    console.log('      WHERE id = auth.uid()');
    console.log('    )');
    console.log('    AND (');
    console.log('      SELECT user_metadata->>\'role\'');
    console.log('      FROM auth.users');
    console.log('      WHERE id = auth.uid()');
    console.log('    ) IN (\'admin\', \'super_admin\', \'member\')');
    console.log('  );');
    console.log('');
    console.log('-- ============================================');
    
    console.log('\n🎯 APÓS APLICAR A MIGRATION:');
    console.log('1. ✅ RLS policies usarão auth.uid() + user_metadata');
    console.log('2. ✅ Sistema compatível com Basic Supabase Authentication');
    console.log('3. ✅ outcomeReasonsApi.ts já está atualizado');
    console.log('4. 🧪 Teste a persistência dos motivos na interface');

  } catch (error) {
    console.error('💥 ERRO:', error.message);
  }

  console.log('\n🏁 SCRIPT CONCLUÍDO');
}

aplicarMigration().then(() => {
  process.exit(0);
}).catch(console.error);

File: apply_migration_direct.py
#!/usr/bin/env python3
"""
🔧 APLICAR MIGRATION: Basic Supabase Authentication
===================================================
"""

import requests
import json
import sys

# Configurações Supabase
SUPABASE_URL = "https://marajvabdwkpgopytvhh.supabase.co"
SERVICE_ROLE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im1hcmFqdmFiZHdrcGdvcHl0dmhoIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc0OTc2NDAwOSwiZXhwIjoyMDY1MzQwMDA5fQ.mkQBVPvhM3OJndsyinoONRUHSDJMh1nFBbPPNH_6cYY"

def execute_sql(sql):
    """Execute SQL via Supabase PostgREST"""
    
    # Headers para service role
    headers = {
        'apikey': SERVICE_ROLE_KEY,
        'Authorization': f'Bearer {SERVICE_ROLE_KEY}',
        'Content-Type': 'application/json',
        'Accept': 'application/json'
    }
    
    # Try different endpoints
    endpoints = [
        f"{SUPABASE_URL}/rest/v1/rpc/exec_sql",
        f"{SUPABASE_URL}/functions/v1/exec_sql", 
        f"{SUPABASE_URL}/rest/v1/exec_sql"
    ]
    
    payload = {"query": sql}
    
    for endpoint in endpoints:
        try:
            print(f"🧪 Tentando endpoint: {endpoint}")
            response = requests.post(endpoint, headers=headers, json=payload, timeout=30)
            print(f"   Status: {response.status_code}")
            
            if response.status_code == 200:
                print("✅ SQL executado com sucesso!")
                return response.json()
            elif response.status_code == 404:
                print("   ❌ Endpoint não encontrado")
                continue
            else:
                print(f"   ⚠️  Resposta: {response.text[:200]}")
                
        except Exception as e:
            print(f"   ❌ Erro: {e}")
            continue
            
    return None

def main():
    print("🔧 APLICANDO MIGRATION: Basic Supabase Authentication")
    print("=" * 55)
    
    # SQL da migration
    migration_sql = """
-- ============================================
-- 🔧 MIGRATION: Basic Supabase Authentication  
-- ============================================

-- ETAPA 1: Remover políticas antigas incompatíveis
DROP POLICY IF EXISTS "Users can view outcome reasons for their tenant" ON pipeline_outcome_reasons;
DROP POLICY IF EXISTS "Admins can manage outcome reasons for their tenant" ON pipeline_outcome_reasons;
DROP POLICY IF EXISTS "Users can view outcome history for their tenant" ON lead_outcome_history;
DROP POLICY IF EXISTS "Users can create outcome history for their tenant" ON lead_outcome_history;

-- ETAPA 2: Criar políticas Basic Supabase Auth
CREATE POLICY "basic_auth_view_outcome_reasons"
  ON pipeline_outcome_reasons
  FOR SELECT
  USING (
    auth.uid() IS NOT NULL
    AND tenant_id = (
      SELECT user_metadata->>'tenant_id'
      FROM auth.users
      WHERE id = auth.uid()
    )
  );

CREATE POLICY "basic_auth_manage_outcome_reasons"
  ON pipeline_outcome_reasons
  FOR ALL
  USING (
    auth.uid() IS NOT NULL
    AND tenant_id = (
      SELECT user_metadata->>'tenant_id'
      FROM auth.users
      WHERE id = auth.uid()
    )
    AND (
      SELECT user_metadata->>'role'
      FROM auth.users
      WHERE id = auth.uid()
    ) IN ('admin', 'super_admin', 'member')
  );

-- ETAPA 3: Políticas para lead_outcome_history
CREATE POLICY "basic_auth_view_outcome_history"
  ON lead_outcome_history
  FOR SELECT
  USING (
    auth.uid() IS NOT NULL
    AND tenant_id = (
      SELECT user_metadata->>'tenant_id'
      FROM auth.users
      WHERE id = auth.uid()
    )
  );

CREATE POLICY "basic_auth_create_outcome_history"
  ON lead_outcome_history
  FOR INSERT
  WITH CHECK (
    auth.uid() IS NOT NULL
    AND tenant_id = (
      SELECT user_metadata->>'tenant_id'
      FROM auth.users
      WHERE id = auth.uid()
    )
    AND (
      SELECT user_metadata->>'role'
      FROM auth.users
      WHERE id = auth.uid()
    ) IN ('admin', 'member', 'super_admin')
  );
"""
    
    # Executar cada statement separadamente para evitar erros de parsing
    statements = [
        'DROP POLICY IF EXISTS "Users can view outcome reasons for their tenant" ON pipeline_outcome_reasons;',
        'DROP POLICY IF EXISTS "Admins can manage outcome reasons for their tenant" ON pipeline_outcome_reasons;',
        'DROP POLICY IF EXISTS "Users can view outcome history for their tenant" ON lead_outcome_history;',
        'DROP POLICY IF EXISTS "Users can create outcome history for their tenant" ON lead_outcome_history;'
    ]
    
    print("\n📋 EXECUTANDO MIGRATION:")
    
    for i, sql in enumerate(statements, 1):
        print(f"\n{i}. {sql[:50]}...")
        result = execute_sql(sql)
        if result is None:
            print(f"   ⚠️  Falha ao executar statement {i}")
        else:
            print(f"   ✅ Statement {i} executado")
    
    print("\n💡 INSTRUÇÕES PARA APLICAR MANUALMENTE:")
    print("=" * 50)
    print("1. Acesse: https://supabase.com/dashboard/project/marajvabdwkpgopytvhh")
    print("2. Vá em: Database > SQL Editor")
    print("3. Execute o arquivo: migration-sql-direct.sql")
    print("\n🎯 PRÓXIMO PASSO: Testar persistência dos motivos!")

if __name__ == "__main__":
    main()